libcontainer/intelrdt/intelrdt.go

```diff
@@ -166,22 +205,38 @@ type Manager interface {
 
 // This implements interface Manager
 type intelRdtManager struct {
-	mu     sync.Mutex
-	config *configs.Config
-	id     string
-	path   string
+	mu              sync.Mutex
+	config          *configs.Config
+	id              string
+	path            string
+	groupType       string
+	switchGroupType func(m *intelRdtManager, getIntelRdtPath func(string, string) (string, error)) error
 }
 
 func NewManager(config *configs.Config, id string, path string) Manager {
+	var groupType string
+	if config.IntelRdt.L3CacheSchema != "" || config.IntelRdt.MemBwSchema != "" {
+		groupType = ControlGroupType
+	} else {
+		groupType = MonitoringGroupType
+	}
 	return &intelRdtManager{
-		config: config,
-		id:     id,
-		path:   path,
+		config:          config,
+		id:              id,
+		path:            path,
+		groupType:       groupType,
+		switchGroupType: switchGroup,
 	}
 }
 
 const (
-	IntelRdtTasks = "tasks"
+	IntelRdtTasks                = "tasks"
+	MonitoringGroupDirectoryName = "mon_groups"
+
+	// See "Resource alloc and monitor groups" part here:
+	// https://www.kernel.org/doc/Documentation/x86/intel_rdt_ui.txt
+	MonitoringGroupType = "MON"
+	ControlGroupType    = "CTRL_MON"
```

Comment:
Could all of them be private?
...
MonitoringGroupType --> monitoringGroupType
ControlGroupType --> controlGroupType